MIGRATION.ru.md

```diff
@@ -1,5 +1,783 @@
+им функциональность выражается через экземпляры элементов .
+
+См. так же изменения про [методы поиска](#Методы-поиска).
+
+Задача: [#581](https://github.com/bem/bem-core/issues/581).
+
+### Методы поиска
+
+Переименованы следующие методы:
+
+- `findBlockInside()` в `findChildBlock()`
+- `findBlocksInside()` в `findChildBlocks()`
+- `findBlockOutside()` в `findParentBlock()`
+- `findBlocksOutside()` в `findParentBlocks()`
+- `findBlockOn()` в `findMixedBlock()`
+- `findBlocksOn()` в `findMixedBlocks()`
+
+Из этих методов удален опциональный первый параметр про элемент.
+
+Добавлены методы: `findChildElem()`, `findChildElems()`, `findParentElem()`, `findParentElems()`, `findMixedElem()`, `findMixedElems()`.
+
+Было:
+
+```js
+this.findBlockInside(this.elem('my-elem'), 'my-block-2');
+```
+
+Стало:
+
+```js
+this.findChildElem('my-elem').findChildBlock(MyBlock2);
+```
+
+Удалены методы: `findElem()`, `closestElem()`, вместо них следует использовать методы `findChildElem()`
+и `findParentElem()`, соответсвенно.
+
+Методы `findChildBlocks()`, `findParentBlocks()`, `findMixedBlocks()`, `findChildElems()`, `findParentElems()`,
+`findMixedElems()` возвращают [коллекции БЭМ-сущностей](#Коллекции).
+
+Методы `findChildElem()` и `findChildElems()` (в отличие от предыдущего аналога `findElem`) не выполняют поиск на собственных DOM-узлах экземпляра.
+
+Было:
+
+```js
+this.findElem('my-elem');
+```
+
+Стало:
+
+```js
+this.findChildElems('my-elem').concat(this.findMixedElems('my-elem'));
+```
+
+Но рекомендется обратить внимание, действительно ли необходимы оба поиска:
+в большинстве случаев достаточно использовать или `this.findChildElems('my-elem')` или `this.findMixedElems('my-elem')`.
+
+#### Проверка вложенности
+
+Вместо удаленного метода `containsDomElem()`, следует использовать метод `containsEntity()`.
+
+Было:
+
+```js
+this.containsDomElem(someElem);
+```
+
+Стало:
+
+```js
+this.containsEntity(someElem);
+```
+
+### Коллекции
+
+Функциональность элемента `collection` блока `i-bem` перестала быть опциональной.
+
+Все методы возвращавшие массив БЭМ-сущностей, теперь возвращают коллекции.
+
+Было:
+
+```js
+this.findBlocksInside('my-block-2')[0].setMod('my-mod', 'my-val');
+```
+
+Стало:
+
+```js
+this.findChildBlocks(MyBlock2).get(0).setMod('my-mod', 'my-val');
+```
+
+Было:
+
+```js
+this.findBlocksInside('my-block-2').forEach(function(myBlock2) {
+    return myBlock2.setMod('my-mod', 'my-val');
+});
+```
+
+Стало:
+
+```js
+this.findChildBlocks(MyBlock2).setMod('my-mod', 'my-val');
+```
+
+Задача: [#582](https://github.com/bem/bem-core/issues/582).
+
+### События
+
+API работы с событиями значильно упрощено. Удалены методы экземпляра блока: `on()`, `un()`, `once()`, `bindTo()`,
+`unbindFrom()`, `bindToDoc()`, `bindToWin()`, `unbindFromDoc()`, `unbindFromWin()` и методы класса: `liveBindTo()`,
+`liveUnbindFrom()`, `on()`, `un()`, `once()`, `liveInitOnBlockEvent()`, `liveInitOnBlockInsideEvent()`.
```

Comment:
What's a proper way to replace `liveInitOnBlockInsideEvent` without callback (like here
https://github.com/bem/bem-components/blob/v3/common.blocks/radio/_type/radio_type_button.js#L37) now?